pkg/cmd/psa/psa.go

```diff
@@ -0,0 +1,346 @@
+`
+
+	empty       = struct{}{}
+	validLevels = map[string]struct{}{
+		string(psapi.LevelPrivileged): empty,
+		string(psapi.LevelBaseline):   empty,
+		string(psapi.LevelRestricted): empty,
+	}
+
+	podControllers = map[string]struct{}{
+		"Deployment":  empty,
+		"DaemonSet":   empty,
+		"StatefulSet": empty,
+		"CronJob":     empty,
+		"Job":         empty,
+	}
+)
+
+// NewCmdPSA creates a new cobra.Command instance that enables checking
+// namespaces for their compatibility with a specified PodSecurity level.
+func NewCmdPSA(parentName string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := PSAOptions{
+		configFlags: &genericclioptions.ConfigFlags{
+			Namespace:  utilpointer.String(""),
+			KubeConfig: utilpointer.String(""),
+		},
+		printFlags: genericclioptions.NewPrintFlags("psa").WithTypeSetter(scheme.Scheme),
+	}
+
+	cmd := cobra.Command{
+		Use:   "psa-check",
+		Short: "Verify namespace workloads match the namespace pod security profile",
+
+		Example: fmt.Sprintf(psaExample, parentName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return fmt.Errorf("validation failed: %w", err)
+			}
+			if err := o.Complete(); err != nil {
+				return fmt.Errorf("completion failed: %w", err)
+			}
+
+			return o.Run()
+		},
+	}
+
+	fs := cmd.Flags()
+	o.configFlags.AddFlags(fs)
+	o.printFlags.AddFlags(&cmd)
+	fs.StringVar(&o.level, "level", "restricted", "The PodSecurity level to check against.")
+	fs.BoolVar(&o.quiet, "quiet", false, "Do not return non-zero exit code on violations.")
+	fs.BoolVarP(&o.allNamespaces, "all-namespaces", "A", o.allNamespaces, "If true, check the specified action in all namespaces.")
+
+	return &cmd
+}
+
+// Validate ensures that all required arguments and flag values are set properly.
+func (o *PSAOptions) Validate() error {
+	if _, ok := validLevels[o.level]; !ok {
+		return fmt.Errorf("invalid level %q", o.level)
+	}
+	return nil
+}
+
+// Complete sets all information required for processing the command.
+func (o *PSAOptions) Complete() error {
+	config := o.configFlags.ToRawKubeConfigLoader()
+	restConfig, err := config.ClientConfig()
+	if err != nil {
+		return fmt.Errorf("failed to create rest config: %w", err)
+	}
+
+	namespace, _, err := config.Namespace()
+	if err != nil {
+		return fmt.Errorf("failed to get namespace: %w", err)
+	}
+	o.namespace = namespace
+
+	// Setup a client with a custom WarningHandler that collects the warnings.
+	o.warnings = &warningsHandler{}
+	restConfig.WarningHandler = o.warnings
+	o.client, err = kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+
+	if o.printFlags.OutputFormat != nil && len(*o.printFlags.OutputFormat) > 0 {
+		printer, err := o.printFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.printObj = printer.PrintObj
+
+		return nil
+	}
+
+	o.printObj = func(object runtime.Object, writer io.Writer) error {
```

Comment:
If we add `.WithDefaultOutput("json")` in https://github.com/openshift/cluster-debug-tools/pull/40/files#diff-a3d5fb2755a1063e47afb3ab8b2b1a436f5433ba352f5eebbcc41505c1ac9dc8R72, we don't need this extra override.

Reply:
That is pretty cool. Thx.